### ISSUE TITLE
Use central connection to ensure that tenant can be created and to create the database

### ISSUE THREAD

**Issue by mattyh88** [archtechx/tenancy#685]
#### Description
Currently the CreateDatabase Job does the following in **handle()**: 

- ensureTenantCanBeCreated() -> Calls **manager()** on the DatabaseConfig class. **manager()** will set "connection" to the returned value of **getTemplateConnectionName()**. **getTemplateConnection()** will return the value for "template_tenant_connection" (from the tenancy config file) if it's not NULL. (We don't set it to NULL, because we need this for other functionalities within the package)
- makeCredentials() -> ignore for now ...
- createDatabase() -> Same here. Calls **manager()** on the DatabaseConfig class and uses the tenant DB connection to create the database. 

Why would we like to create a database with the tenant DB user credentials? We would like to create the database with the central 'admin' connection (for which the configured DB user has more permissions). 

Long story short: We would only like to grant SELECT, INSERT, UPDATE, DELETE for the tenant DB user. Thus, the ensureTenantCanBeCreated() and createDatabase() functions should use the central 'admin' connection by default.

#### Why this should be added
Better security as the tenant DB user doesn't need too much permissions. 

Thank you :)

**Comment by stancl** [comment]
> tenant DB user doesn't need too much permissions

Why not use https://github.com/archtechx/tenancy/blob/3.x/src/TenantDatabaseManagers/PermissionControlledMySQLDatabaseManager.php?

**Comment by mattyh88** [comment]
Will that change the fact that the DB user - which is used to "check if a certain tenant DB exists already before creating it" & "to actually create the tenant DB" - is the one that is used for the central 'admin' connection? I thought that the PermissionControlledMySQLDatabaseManager class just manages the tenant DB users (create DB user, delete DB user, ...)

**Comment by roman-savuliak** [comment]
+1.  That is strange that DB user that should theoretically have access only to tenant DB can query information_schema and create new databases. For sure this should be done from central connection only

**Comment by stancl** [comment]
> That is strange that DB user that should theoretically have access only to tenant DB can query information_schema and create new databases

Does the user do that? I think the central connection is used for creating and deleting databases? And the tenant DB user can only manage tables inside that tenant's database.

**Comment by F-Georg** [comment]
Sorry for coming back to this, however I had a problem in a project, which is directly related to the content of mathy88's post, with which he started this issue. And I would like to ask for help.

Is it ok, if I write my post in this issue or shall I open a new issue?

First of all many thanks for providing this great package!

The issue in my project is the following.
I have two database connections.
The database user of "admin_connection" has all the MySQL privileges, like among others Create.
I use this connection in config/tenancy.php as 'central_connection'.
The database user of "template_connection" only has the Select, Insert, Update and Delete MySQL privileges.
I use this connection in config/tenancy.php as 'template_tenant_connection'.

If I insert a tenant into the tenant table, for example with the tenant name "example_name", I get this error:
`SQLSTATE[42000]: Syntax error or access violation: 1044 Access denied for user 'db_user_of_template_connection'@'localhost' to database 'example_name' (SQL: CREATE DATABASE `example_name` CHARACTER SET `utf8mb4` COLLATE `utf8mb4_unicode_ci``

I then found this issue on Github and looked at the files, that mattyh88 mentioned.

When creating a tenant database, in the handle function of CreateDatabase.php the following is executed:
`$this->tenant->database()->manager()->createDatabase($this->tenant);`
If I understood it right, in DatabaseConfig.php in the function manager() a database connection is selected through the function getTemplateConnectionName() in DatabaseConfig.php.
The function getTemplateConnectionName() returns the internal database connection of the tenant.
If there is no internal database connection of the tenant, the template_tenant_connection is returned and if there is no template_tenant_connection either, the central_connection is returned.

In my case there is no internal database connection of the tenant. Therefore the template_tenant_connection is used for creating the tenant database.
However in my case the template_tenant_connection doesn't have the necessary privileges for creating databases and I would like to keep it that way.

Is there a way to create a tenant database, if one wants to grant the privilege for creating databases only to the db user of the central_connection, but not to the db user of the internal database connection and not to the db user of the template_tenant_connection?

Many thanks in advance for your help.

**Comment by stancl** [comment]
Could you create a separate issue for that? I think that'd be best

**Comment by F-Georg** [comment]
@stancl  
Yes, sure!
I opened a new issue: [https://github.com/archtechx/tenancy/issues/806](https://github.com/archtechx/tenancy/issues/806)
